### ISSUE TITLE
SFTP PHP 5.6 using phpseclib hangs

### ISSUE THREAD

**Issue by vlopez** [phpseclib/phpseclib#711]
Could someone please help?
I have been at this all day, can't make it work.

http://stackoverflow.com/questions/30992629/sftp-php-5-6-using-phpseclib-hangs

It shouldn't be this hard.

**Comment by terrafrost** [comment]
Honestly, my guess is that it doesn't have anything to do with PHP 5.6 but rather the window size. I'd (1) try the latest git version of phpseclib. If that doesn't work for you lmk. The link:

https://github.com/phpseclib/phpseclib/archive/1.0.zip

Here's the specific commit that I think may fix the issue for you:

https://github.com/phpseclib/phpseclib/pull/538

I don't have time to look through the logs right now. I'm at my regular 8-5 day job right now and altho I can spend some time making posts like this I can't spend my entire day diagnosing issues and what not.

**Comment by vlopez** [comment]
I was actually using the namespace version one I downloaded from github.
Anyway I tried the version from your link (new Net_SFTP) but it still hangs, same kind of log.

Any help is welcomed. Thank you

**Comment by terrafrost** [comment]
From your log file:

```
<- NET_SSH2_MSG_CHANNEL_OPEN_CONFIRMATION (since last: 0.1995, network: 0.0001s)
00000000  00:00:01:00:00:00:00:00:ff:ff:ff:ff:00:00:80:00  ................
```

The initial window size is 0xFFFFFFFF. Since PHP uses signed integers that gets read as -1. eg.

```php
extract(unpack('Nsize', "\xFF\xFF\xFF\xFF"));
```

That'll produce a variable - `$size` - that's equal to -1.

Anyway, try this:

https://github.com/terrafrost/phpseclib/archive/neg_window_size_fix-1.0.zip

The commit:

https://github.com/terrafrost/phpseclib/commit/c0555581e755b68d0d2019a5fb2a7b15f2fff50a

Thanks!

**Comment by vlopez** [comment]
I see... wow ok.
That worked.

I was trying to understand that line...

So if this was a x64 PHP installation that would not be an issue?

Thank you!

**Comment by terrafrost** [comment]
Yup - exactly. Quoting ...,

> 64-bit platforms usually have a maximum value of about 9E18, except for Windows, which is always 32 bit.

So my guess is that you're running Windows but maybe you have a weird Linux install too idk.

Like on an Ubuntu 15.04 machine `extract(unpack('Nsize', "\xFF\xFF\xFF\xFF"))` produces 4,294,967,295 but on my Windows 8.1 laptop I get -1.

**Comment by bantu** [comment]
This doesn't seem to have anything to do with Windows in particular. The problem is related to 32-bit PHP platforms in general. The "Caution" notes on http://php.net/manual/en/function.unpack.php apply.

**Comment by bantu** [comment]
We are claiming this was fixed by #717. Closing as such.
